Api

Matribuk opened the pull request:
# Description

Addition of an API with variable management routes on a vault.
Resolving part of issue : "[Feature]: Create an API to interact with the Vault"

# Changes include

- [ ] Ability to call the vault to manage project and project variables

# Additional comments

Check the router.go file to see how to call the api and don't forget "export TOKEN_VPS_VAULT".

Matribuk committed:
feat(api): var abilities & core api

Matribuk committed:
feat(api): add routes var

Matribuk committed:
feat(api): add routes project

RezaRahemtola commented on api/.gitignore:
```diff
@@ -0,0 +1 @@
+go.sum
```
why did you put this here ?

RezaRahemtola commented on api/controllers/config.go:
Should be an environment variable

RezaRahemtola commented on api/controllers/variable.go:
```diff
@@ -0,0 +1,160 @@
+		return "error client auth"
+	}
+
+	client.SetToken(token)
```
Duplicated code between controllers, you could create a function / middleware for this

RezaRahemtola commented on api/controllers/variable.go:
```diff
@@ -0,0 +1,160 @@
+	client.SetToken(token)
+
+	// Ask the engine version of the project
+	var temp_json = List_vars(name_project)
```
```suggestion
	temp_json := List_vars(name_project)
```

RezaRahemtola commented on api/routes/variable.go:
```diff
@@ -0,0 +1,48 @@
+	name_project := c.Param("project")
+	name_var := c.Param("variable")
+	response := controllers.Edit_vars(name_project, name_var, myVar.Value)
+	c.JSON(200, gin.H{
+		"message": response,
```
You always send a 200 response code event if the request failed

RezaRahemtola commented on api/routes/variable.go:
```diff
@@ -0,0 +1,48 @@
+	name_project := c.Param("project")
+	name_var := c.Param("variable")
+	response := controllers.Edit_vars(name_project, name_var, myVar.Value)
+	c.JSON(200, gin.H{
```
Also use statuses from `net/http` for better code clarity https://go.dev/src/net/http/status.go

Matribuk committed:
feat(api): allow the api to respond with proper http status and add error handling
BREAKING CHANGE: environment key ADDRESS_VPS_VAULT in config.go file in middlewares folder is needed

Matribuk committed:
feat(api): add project routes

Matribuk committed:
feat(api): README.md created

Matribuk committed:
fix(api): README.md edit

Matribuk committed:
fix(api): resolve review notes of PR
